azurelinuxagent/common/event.py

```diff
@@ -341,12 +335,30 @@ def add_metric(self, category, counter, instance, value, log_event=False):
         event.parameters.append(TelemetryEventParam('Instance', instance))
         event.parameters.append(TelemetryEventParam('Value', value))
 
+        self._add_default_parameters_to_event(event)
         data = get_properties(event)
         try:
             self.save_event(json.dumps(data))
         except EventError as e:
             logger.error("{0}", e)
 
+    @staticmethod
+    def _add_default_parameters_to_event(event):
+        task_name = threading.current_thread().getName()
+
+        # We set thread names for the threads spawned by ExtHandler. We get the name of the thread,
```

Comment:
where do we set the thread name?

Reply:
Defined in the of MonitorHandler and EnvHandler `start()`.

`self.server_thread.setName("EnvHandler")` ([link](https://github.com/Azure/WALinuxAgent/pull/1663/files#diff-e553d83092bcdeb28abdda5cbb040a9aR91))
`self.event_thread.setName("MonitorHandler")` ([link](https://github.com/Azure/WALinuxAgent/pull/1663/files#diff-3d8dc864f0284d65cf57fb2b2f9799a0R165)).

Reply:
the we should respect those values instead of setting them @352, right?

Reply:
It does. In @352.

```
test_suite = ['EnvHandler', 'MonitorHandler', 'MainThread']

for i in test_suite:
    task_name = i
    task_name = task_name if task_name != "MainThread" else "ExtHandler"
    print(task_name)
```
Output:
```
/usr/bin/python3.6 /home/varad/.PyCharm2019.2/config/scratches/scratch_3.py
EnvHandler
MonitorHandler
ExtHandler
```

Reply:
I mean respect whatever value the name has, instead of changing it @352